include/zenoh-pico/collections/refcount.h

```diff
@@ -38,122 +41,292 @@
+        name##_inner_rc_t *in;                                                                  \
+    } name##_weak_t;                                                                            \
+                                                                                                \
+    static inline name##_rc_t name##_rc_null(void) {                                            \
+        name##_rc_t p;                                                                          \
+        p.in = NULL;                                                                            \
+        return p;                                                                               \
+    }                                                                                           \
+    static inline name##_rc_t name##_rc_new(void) {                                             \
+        name##_rc_t p;                                                                          \
+        p.in = (name##_inner_rc_t *)z_malloc(sizeof(name##_inner_rc_t));                        \
+        if (p.in != NULL) {                                                                     \
+            memset(&p.in->val, 0, sizeof(type##_t));                                            \
+            _ZP_RC_OP_INIT_CNT                                                                  \
+        }                                                                                       \
+        return p;                                                                               \
+    }                                                                                           \
+    static inline name##_rc_t name##_rc_new_from_val(type##_t val) {                            \
+        name##_rc_t p;                                                                          \
+        p.in = (name##_inner_rc_t *)z_malloc(sizeof(name##_inner_rc_t));                        \
+        if (p.in != NULL) {                                                                     \
+            p.in->val = val;                                                                    \
+            _ZP_RC_OP_INIT_CNT                                                                  \
+        }                                                                                       \
+        return p;                                                                               \
+    }                                                                                           \
+    static inline name##_rc_t name##_rc_clone(const name##_rc_t *p) {                           \
+        name##_rc_t c;                                                                          \
+        unsigned int cmp_val = UINT32_MAX;                                                      \
+        if (_ZP_RC_OP_CHECK_WEAK_CNT) {                                                         \
```

Comment:
this check will not work in mt scenario since in case of simultaneous clones of 2 orcs, both of them can pass this check before increasing weak count in the end of the function.